apps/hash-graph/lib/graph/src/api/rest.rs

```diff
@@ -111,16 +111,33 @@ pub struct RestRouterDependencies<P: StorePool + Send + 'static> {
     pub domain_regex: DomainValidator,
 }
 
+/// A [`Router`] that only serves the OpenAPI specification (JSON, and necessary subschemas) for
+/// the REST API.
+pub fn openapi_only_router() -> Router {
+    let open_api_doc = OpenApiDocumentation::openapi();
+
+    Router::new().nest(
+        "/api-doc",
+        Router::new()
+            .route(
+                "/openapi.json",
+                get({
+                    let doc = open_api_doc;
+                    move || async { Json(doc) }
+                }),
+            )
```

Comment:
```suggestion
            .route("/openapi.json", get(|| async { Json(open_api_doc) }))
```

Reply:
Updating all of our existing code in this PR ey 😅